schema/omega-services.async.api.yaml

```diff
@@ -1,525 +1,415 @@
+        the `OMGToken` header.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/updateResponseStatus"
-      correlationId:
-        description: Default Correlation ID
-        location: $message.header#/correlationId
-
-    errorResponse:
-      name: errorResponse
-      title: Error response message
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    UnrecognisedMessageTypeError:
+      messageId: OELIFERR005
+      name: UnrecognisedMessageTypeError
+      title: Unrecognised Message Type Error
+      description: >-
+        The value of `OMGMessageTypeID` header was not recognised and so the
+        message cannot be further dispatched for processing.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
+      payload:
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    InvalidMessageError:
+      messageId: OELIFERR006
+      name: InvalidMessageError
+      title: Invalid Message Error
+      description: >-
+        The content of the body of the message has failed validation, e.g. a
+        parsed JSON message may be missing some required properties, or those
+        properties may have invalid values.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
+      payload:
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+    ProcessingError:
+      messageId: OELIFERR007
+      name: ProcessingError
+      title: ProcessingError
+      description: >-
+        The received message was valid, but the system was unable complete the
+        processing of a message.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/errorMessage"
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+  messageTraits:
+    JMSMessage:
+      name: JMS Message
+      description: Headers that are mostly common to all JMS Messages
+      headers:
+        required:
+          - JMSMessageID
+          - JMSTimestamp
+        properties:
+          JMSMessageID:
+            name: JMSMessageID
+            description: >-
+              A unique identifier for the message. We suggest using a short
+              application instance specific prefix followed by a unique
+              identifier. In the extreme case a UUID may be used.
```

Comment:
Isn't the JMS message ID assigned by SQS in our case?

Reply:
It is typical for both the `JMSMessageID` and `JMSTimestamp` headers to be assigned by the provider, some implementations allow you to override this but not all.

Do you see a problem with this?

Reply:
I'm not sure what the SQS policy is with setting the `JMSTimestamp`, I only know about the `JMSMessageID` because of that comment in the Amazon code and the fact that I've tried setting it and it comes out as something else. I guess it doesn't matter to us, as long as it gets set, and that's what this schema is saying. Maybe it would be worth adding a comment along the lines of _these headers may be set by your JMS implementation_.